lisa/runners/lisa_runner.py

```diff
@@ -116,18 +116,18 @@ def fetch_task(self) -> Optional[Task[None]]:
         # Loop environments instead of test results, because it needs to reuse
         # environment as much as possible.
         if available_results and available_environments:
-            for priority in range(6):
-                can_run_results = self._get_results_by_priority(
-                    available_results, priority
-                )
-                if not can_run_results:
+            # it means there are test cases and environment, so it needs to
+            # schedule task.
+            for environment in available_environments:
+                if environment.is_in_use:
+                    # skip in used environments
                     continue
 
-                # it means there are test cases and environment, so it needs to
-                # schedule task.
-                for environment in available_environments:
-                    if environment.is_in_use:
-                        # skip in used environments
+                for priority in range(6):
+                    can_run_results = self._get_results_by_priority(
```

Comment:
We can totally remove the `by_priority` concept, because it's not valuable as a second level loop. After it's removed, the logic is simpler.

Reply:
I think it is still important for priority order to be followed as much as possible, but I believe this is achieved by sorting the results beforehand, correct?

Reply:
Yes, the order of results can achieve it too.